snuba/manual_jobs/recreate_missing_eap_spans_materialized_views.py

```diff
@@ -0,0 +1,29 @@
+    "CREATE MATERIALIZED VIEW IF NOT EXISTS default.spans_str_attrs_3_mv TO default.spans_str_attrs_3_local (`organization_id` UInt64, `project_id` UInt64, `attr_key` String CODEC(ZSTD(1)), `attr_value` String CODEC(ZSTD(1)), `timestamp` DateTime CODEC(DoubleDelta, ZSTD(1)), `retention_days` UInt16, `count` SimpleAggregateFunction(sum, UInt64)) AS SELECT organization_id, project_id, attrs.1 AS attr_key, attrs.2 AS attr_value, toStartOfDay(_sort_timestamp) AS timestamp, retention_days, 1 AS count FROM default.eap_spans_2_local LEFT ARRAY JOIN arrayConcat(CAST(attr_str_0, 'Array(Tuple(String, String))'), CAST(attr_str_1, 'Array(Tuple(String, String))'), CAST(attr_str_2, 'Array(Tuple(String, String))'), CAST(attr_str_3, 'Array(Tuple(String, String))'), CAST(attr_str_4, 'Array(Tuple(String, String))'), CAST(attr_str_5, 'Array(Tuple(String, String))'), CAST(attr_str_6, 'Array(Tuple(String, String))'), CAST(attr_str_7, 'Array(Tuple(String, String))'), CAST(attr_str_8, 'Array(Tuple(String, String))'), CAST(attr_str_9, 'Array(Tuple(String, String))'), CAST(attr_str_10, 'Array(Tuple(String, String))'), CAST(attr_str_11, 'Array(Tuple(String, String))'), CAST(attr_str_12, 'Array(Tuple(String, String))'), CAST(attr_str_13, 'Array(Tuple(String, String))'), CAST(attr_str_14, 'Array(Tuple(String, String))'), CAST(attr_str_15, 'Array(Tuple(String, String))'), CAST(attr_str_16, 'Array(Tuple(String, String))'), CAST(attr_str_17, 'Array(Tuple(String, String))'), CAST(attr_str_18, 'Array(Tuple(String, String))'), CAST(attr_str_19, 'Array(Tuple(String, String))'), [('sentry.service', service), ('sentry.segment_name', segment_name), ('sentry.name', name)]) AS attrs GROUP BY organization_id, project_id, attr_key, attr_value, timestamp, retention_days",
+]
+
+
+class RecreateMissingEAPSpansMaterializedViews(Job):
+    def __init__(self, job_spec: JobSpec) -> None:
+        super().__init__(job_spec)
+
+    def execute(self, logger: JobLogger) -> None:
+        cluster = get_cluster(StorageSetKey.EVENTS_ANALYTICS_PLATFORM)
+
+        for storage_node in cluster.get_local_nodes():
+            connection = cluster.get_node_connection(
+                ClickhouseClientSettings.CLEANUP,
+                storage_node,
+            )
+            for query in materialized_views:
+                logger.info("Executing query: {query}")
+                connection.execute(query=query)
```

Comment:
add `mutations_sync=0` in the settings so you don't wait on the operation to complete

Reply:
I had it at first but the documentation specifically says it's for `ALTER` mutations and somehow, `CREATE` doesn't make the cut so I don't think it's very useful.

https://clickhouse.com/docs/en/operations/settings/settings#mutations_sync